### PR TITLE
ignores table options on ALTER TABLE

### PR DESCRIPTION
**Pull request by rudolfix**
### Description
bigquery destination was raising when table was ALTERed and table options are present. this failed repeated runs of pipelines using `bigquery_adapter` that were migrating schemas.

### CODE REVIEW

File: tests/load/bigquery/test_bigquery_table_builder.py [hunk 1]
```diff
@@ -863,6 +869,16 @@ def sources() -> List[DltResource]:
         assert not no_hints_table.description
         assert hints_table.description == "Once upon a time a small table got hinted."
 
+    # run sources() again to make sure that table hints are ignored on alter table
```
Comment (Pipboyguy) [inline tests/load/bigquery/test_bigquery_table_builder.py:872]: Reading this in isolation suggests that the intended behaviour is to not perform the ALTER migration for OPTION hint.

I suggest we leave the original `test_adapter_additional_table_hints_table_description` test and add `test_adapter_additional_table_hints_table_description_with_alter_table` as separate one, with an assert like

```python
assert hints_table.description == "Once upon a time a small table got hinted twice.",
``` 

We can then mark this test as `skip`  until support is added.
Let me know what you think!

File: tests/load/bigquery/test_bigquery_table_builder.py [hunk 2]
```diff
@@ -887,7 +903,7 @@ def no_hints() -> Iterator[Dict[str, str]]:
         yield from [{"col1": str(i)} for i in range(10)]
 
     hints = bigquery_adapter(
-        no_hints._clone(new_name="hints"), table_expiration_datetime="2030-01-01"
+        no_hints.with_name(new_name="hints"), table_expiration_datetime="2030-01-01"
```
Comment (Pipboyguy) [inline tests/load/bigquery/test_bigquery_table_builder.py:906]: Thanks for all these improvements!

**Review by Pipboyguy — CHANGES_REQUESTED** [review]
Nice!

I made one or two comments for your consideration.

File: dlt/destinations/impl/bigquery/bigquery.py [hunk 3]
```diff
@@ -337,7 +337,10 @@ def _get_table_update_sql(
             return sql
 
         if generate_alter:
-            raise NotImplementedError("Update table options not yet implemented.")
+            logger.info(
```
[the remark this answers is not in the document]
Comment (rudolfix) [reply dlt/destinations/impl/bigquery/bigquery.py:340]: I considered that. but unfortunately this is a "normal" behavior - ie you declare a table with a description and then you'll get a warning each time the schema is updated. we'd need to check if the description is there - and if it isn't we could issue a warning. but this is out of scope of the ticket - I just want to fix a bug

[the remark this answers is not in the document]
Comment (Pipboyguy) [reply dlt/destinations/impl/bigquery/bigquery.py:340]: cool makes sense

**Review by Pipboyguy — APPROVED** [review]

**Pull request merged**